test/e2e/jobs/jobs.test.ts

```diff
@@ -0,0 +1,166 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { GenericContainer, Network, StartedNetwork, StartedTestContainer, TestContainers, Wait } from "testcontainers";
+// import { LogWaitStrategy } from "testcontainers/build/wait-strategies/log-wait-strategy";
+import { CommunicationProtocolEnum, DaprClient, DaprServer } from "../../../src";
+import { scheduler } from "node:timers/promises";
+// import { AbstractWaitStrategy } from "testcontainers/build/wait-strategies/wait-strategy";
+
+jest.setTimeout(10000);
+
+describe("Jobs End to End", () => {
+
+    let network: StartedNetwork | null = null;
+    let daprScheduler: StartedTestContainer | null = null;
+    let daprd: StartedTestContainer | null = null;
+    let server: DaprServer | null = null;
+    let client: DaprClient | null = null;
+
+    beforeAll(async () => {
+
+        await TestContainers.exposeHostPorts(8070);
+
+        network = await new Network().start();
+
+        daprScheduler = await new GenericContainer("ghcr.io/dapr/dapr")
+          .withName("dapr-js-sdk-test-scheduler")
+          .withNetwork(network)
+          .withNetworkAliases("scheduler")
+          .withExposedPorts(8083)
+          .withCommand([
+            "./scheduler",
+            // note: Don't think this is necessary, buuuuut????
+            "--listen-address", "0.0.0.0",
+            "--port", "8083",
+            "--log-level", "debug",
+            // note: This feels redundant, but here as yet another thing I've tried.
+            // "--mode", "standalone",
+          ])
+          .withTmpFs({
+            "/data": "rw",
+          })
+          // note: Because dapr containers don't have `sh` or `bash` inside, this is kind of the best health check.
+          .withWaitStrategy(Wait.forLogMessage("api is ready").withStartupTimeout(10000))
+          .start();
+
+        daprd = await new GenericContainer("ghcr.io/dapr/dapr")
+          .withName("dapr-js-sdk-test-daemon")
+          .withNetwork(network)
+          .withNetworkAliases("daprd")
+          .withExposedPorts(8081, 8082)
+          .withCommand([
+            "./daprd",
+            "--app-id", "dapr-js-sdk-testing",
+            // todo: Need to figure out how to tell daprd where my app can be found as it's not on `localhost`
+            "--app-channel-address", "host.testcontainers.internal",
+            "--app-port", "8070",
+            "--dapr-grpc-port", "8081",
+            "--dapr-http-port", "8082",
+            "--scheduler-host-address", `scheduler:${getPort(daprScheduler, 8083)}`,
```

Comment:
If they are both running on the same network, why use the host port?

```suggestion
            "--scheduler-host-address", `scheduler:8083`,
```